noggin/controller/user.py

```diff
@@ -188,3 +189,49 @@ def user_settings_otp_add(ipa, username):
             flash(error, 'danger')
 
     return redirect(url_for('user_settings_otp', username=username))
+
+@app.route('/user/<username>/settings/otp/disable/', methods=['POST'])
+@with_ipa(app, session)
+def user_settings_otp_disable(ipa, username):
+    # TODO: Maybe make this a decorator some day?
+    if session.get('noggin_username') != username:
+        flash('You do not have permission to edit this account.', 'danger')
+        return redirect(url_for('user', username=username))
+
+    form = UserSettingsDisableOTPForm()
+    user = User(user_or_404(ipa, username))
+
+    if form.validate_on_submit():
+        username = session.get('noggin_username')
+        token = form.token.data
+
+        tokens = [
+            OTPToken(t)
+            for t in ipa._request(
+                'otptoken_find', [], {'ipatokenowner': username, 'all': True}
+            )['result']
+        ]
+        enabled_count = 0
+        for otp in tokens:
+            if not otp.disabled:
+                enabled_count += 1
+                if enabled_count > 1: break
+
+        if enabled_count > 1:
```

Comment:
Do you think we could move this logic to the otp list page, then not show the form if there is only 1 enabled otp key?

Pretty sure IPA won't let us disable the last token, even if the user tries to submit an outdated form.